### PR TITLE
Fixes #17 Add workflow for R package website building

### LINKED ISSUE #17: Add workflow for R package website building

### PR DESCRIPTION
**Pull request by Yuri05**

### CODE REVIEW

File: .github/workflows/CreateGitHubPagesForR.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
+name: Create GitHub pages for an R package
+
+on: 
+  workflow_call:
+    inputs:
+      pkgdown_mode:
+        required: true
+        type: string
```
Comment (Yuri05) [inline .github/workflows/CreateGitHubPagesForR.yml:8]: must be one of 'devel' or 'release'
Unfortunately it seems to be no possibility to directly check the input enum values (the [choice type](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onworkflow_dispatchinputsinput_idtype) is not available for this type of workflow - not sure why)

File: .github/workflows/CreateGitHubPagesForR.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      extra_r_packages:
+        required: false
+        type: string
```
Comment (Yuri05) [inline .github/workflows/CreateGitHubPagesForR.yml:11]: additional r packages to be installed, separated by comma

File: .github/workflows/CreateGitHubPagesForR.yml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+      gh-pages-branch:
+        required: false
+        type: string
+        default: 'gh-pages'
```
Comment (Yuri05) [inline .github/workflows/CreateGitHubPagesForR.yml:15]: branch for deploying of gh pages. 'gh-pages' as per default

**Review by pchelle — APPROVED** [review]
Nice !
If I underderstand, we can replace most of the current yaml by the code below ?

```yaml
jobs:
  build-website:
    uses: Open-Systems-Pharmacology/Workflows/.github/workflows/CreateGitHubPagesForR.yml@main
    with:
      pkgdown_mode: devel
#    extra_r_packages: 
#    gh-pages-branch:

```

**Pull request merged**